### PR TITLE
Rules: added rudimentary phone number validator

### PR DESCRIPTION
**Pull request by mlebrun**

### CODE REVIEW

File: src/Validation/Rules/PhoneRule.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+/**
+ * Validates if a phone number is valid or not.
+ *
+ * NOTE: this rule cannot extend \Behance\NBD\Validation\Abstracts\RegexRuleAbstract
+ *       as it needs to use `preg_match_all` instead of missing `g` regex option
+ *
+ * @see  http://stackoverflow.com/questions/3578671/unknown-modifier-g-in-when-using-preg-match-in-php
+ */
+class PhoneRule extends RuleAbstract {
```
Comment (bryanlatten) [inline src/Validation/Rules/PhoneRule.php]: isn't there a regex abstract?

File: tests/Validation/Rules/PhoneRuleTest.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        [ 'expected' => true,  'data' => '202 555 4567' ],
+        [ 'expected' => true,  'data' => '4035555678' ],
+        [ 'expected' => true,  'data' => '1 416 555 9292' ],
+        [ 'expected' => false, 'data' => '926 3 4' ],
+        [ 'expected' => false, 'data' => '8 800 600-APPLE' ],
+        [ 'expected' => false, 'data' => '+-()--' ],
+        [ 'expected' => false, 'data' => '+1(234)--(54)' ],
+        [ 'expected' => false, 'data' => '+1(234--4)' ],
+        [ 'expected' => false, 'data' => '+42 555.123).4567' ],
+    ];
```
Comment (markdunphy) [inline tests/Validation/Rules/PhoneRuleTest.php]: what aboutttt (123)-456-7890

**Comment by markdunphy** [comment]
LGTM

**Comment by mikeklein13** [comment]
:+1:

File: tests/Validation/Rules/PhoneRuleTest.php (hunk 2 above)
Comment (mlebrun) [inline tests/Validation/Rules/PhoneRuleTest.php]: added

File: tests/Validation/Rules/PhoneRuleTest.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  } // isValid
+
+  /**
+   * @return array
+   */
+  public function testDataProvider() {
+
+    return [
+        [ 'expected' => true,  'data' => '+42 555.123.4567' ],
```
Comment (bryanlatten) [inline tests/Validation/Rules/PhoneRuleTest.php]: please add test case for bools, closures, objects, null

**Comment by bryanlatten** [comment]
:+1:
